src/psyclone/gocean1p0.py

```diff
@@ -204,16 +192,44 @@ def gen_code(self, parent):
                        content.
         :type parent: `psyclone.f2pygen.ModuleGen`
         '''
-        # First check if there is any unsupported invoke, in this case the
-        # f2pygen gen_code path is used instead
+        opencl_kernels = []
         for invoke in self.invoke_list:
+
             # TODO 1134: The opencl path is still largely implemented using
-            # the f2pygen and cannot be processed by the backend yet.
-            if invoke.schedule.opencl:
-                super(GOInvokes, self).gen_code(parent)
-                return
+            # the f2pygen and cannot be processed by the backend.
+            if isinstance(invoke.schedule, GOInvokeSchedule) and \
+                    invoke.schedule.opencl:
+                name = invoke.schedule.name
+                temporary_module = ModuleGen("dummy")
+
+                # Generate set_args in a temporary fparser module
+                for kern in invoke.schedule.coded_kernels():
+                    if kern.name not in opencl_kernels:
+                        opencl_kernels.append(kern.name)
+                        kern.gen_arg_setter_code(temporary_module)
+
+                # Generate invoke in a temporary fparser module
+                invoke.gen_code(temporary_module)
+
+                # Add the invoke and set_args subroutines as CodeBlocks in
+                # the PSyIR tree representing the PSy layer.
+                for item in temporary_module.root.content:
+                    if hasattr(item, 'name') and \
+                            (item.name == name or
+                             item.name.endswith("set_args")):
+                        reader = FortranStringReader(str(item))
+                        reader.set_format(FortranFormat(True, True))
+                        sub = Subroutine_Subprogram(reader)
+                        codeblock = CodeBlock(
+                            [sub], CodeBlock.Structure.STATEMENT)
+                        if item.name == name:
+                            invoke.schedule.replace_with(codeblock)
+                            invoke.schedule = codeblock
+                        else:
+                            invoke.schedule.parent.addchild(codeblock)
 
             # TODO 1168: PSyDataNodes are not supported by the backend yet.
+            # For now the f2pygen gen_code path is used instead
             if invoke.schedule.root.walk(PSyDataNode):
                 super(GOInvokes, self).gen_code(parent)
                 return
```

Comment:
pylint points out that the `invoke` loop variable is useed at line 239 and 244 which may not be correct?

Reply:
Fixed